rubin_sim/sim_archive/sim_archive.py

```diff
@@ -0,0 +1,1042 @@
+            capture_env=True,
+            opsim_metadata=opsim_metadata,
+        )
+
+        if archive_uri is not None:
+            resource_path = transfer_archive_dir(data_dir.name, archive_uri)
+        else:
+            resource_path = ResourcePath(data_dir.name, forceDirctory=True)
+
+    results = sim_results + (resource_path,)
+    return results
+
+
+def compile_sim_archive_metadata_cli(*args):
+    parser = argparse.ArgumentParser(description="Create a metadata compilation HDF5 file at a URI")
+    parser.add_argument(
+        "--compilation_uri",
+        type=str,
+        default=None,
+        help="The URI of the metadata archive compilation to write. "
+        + "Defaults to compilation_metadate.h5 in the archive.",
+    )
+    parser.add_argument(
+        "--archive_base_uri",
+        type=str,
+        default="s3://rubin:rubin-scheduler-prenight/opsim/",
+        help="Base URI for the archive",
+    )
+    parser.add_argument(
+        "--append",
+        action="store_true",
+        help="Do not rebuild the whole compilation, "
+        + "but add new simulations with dates after the last current entry.",
+    )
+
+    arg_values = parser.parse_args() if len(args) == 0 else parser.parse_args(args)
+    archive_uri = arg_values.archive_base_uri
+    compilation_uri = arg_values.compilation_uri
+    append = arg_values.append
+    if compilation_uri is None:
+        compilation_resource = ResourcePath(archive_uri).join("compiled_metadata_cache.h5")
+    else:
+        compilation_resource = ResourcePath(compilation_uri)
+
+    compilation_resource = compile_sim_metadata(archive_uri, compilation_resource, append=append)
+
+
+def fetch_prenight_visits(
+    day_obs: str | None = None,
+    tags: tuple[str] = ("ideal", "nominal"),
+    max_simulation_age: int = 2,
+    archive_uri: str = "s3://rubin:rubin-scheduler-prenight/opsim/",
+    compilation_uri: str = "s3://rubin:rubin-scheduler-prenight/opsim/compiled_metadata_cache.h5",
+    **kwargs,
+) -> pd.DataFrame:
+    """Fetches visit parameters from the latest archived pre-night simulation.
+
+    Parameters
+    ----------
+    day_obs : `str` or  `None`
+        The date of the evening for the night for which to get visits.
+        If `None`, then the current date will be used.
+    tags : `tuple[str]`
+        A tuple of tags to filter simulations by.
+        Defaults to ``('ideal', 'nominal')``.
+    max_simulation_age : `int`
+        The maximum age of simulations to consider, in days. Simulations older
+        than ``max_simulation_age`` will not be considered. Defaults to 2.
+    archive_uri : `str`
+        The URI of the archive from which to fetch the simulation.
+        Defaults to ``s3://rubin:rubin-scheduler-prenight/opsim/``.
+    compilation_uri : `str`
+        The URI of the compiled metadata HDF5 file for efficient querying.
+        Defaults to ``s3://rubin:rubin-scheduler-prenight/opsim/compiled_metadata_cache.h5``.
+
```

Comment:
I think it's helpful to have some brief explanation of what the kwargs can be or what they're used for. 
Even 
```
kwargs : `dict`
     kwargs are passed to rubin_sim.maf.get_sim_data, allowing selection of columns or constraints.
```
would give some guidance. 
(if kwargs are typically documented in a Notes section or something else, in numpydoc standard, I'd do it like that instead).